zcfan.c

```diff
@@ -225,6 +225,16 @@ static void maybe_ping_watchdog(void) {
         }                                                                      \
     } while (0)
 
+#define fscanf_str_for_key(f, pos, name, dest)                                 \
+    do {                                                                       \
+        char val[256];                                                         \
+        if (fscanf(f, name " %255s ", val) == 1) {                             \
+            dest = strdup(val);                                                \
```

Comment:
To make this LSAN clean you'll also need to free this later. You can do this by having a struct with another member, `bool alloc`.

Reply:
Even more simply, you can just make `struct Rule`'s tcpacpi_level an array instead of a pointer, which avoids needing heap allocation at all.